### ISSUE TITLE
Race condition with projects and outputs

### ISSUE THREAD

**Issue by Stono** [terraform-providers/terraform-provider-google#1686]
Hi There, 
I'm looking for some advice to make this collection of resources work.  We're basically trying to:

- Setup a new project
- Enable compute API
- Get the default service account email address

```
resource "google_project" "project" {
...
}

resource "google_project_services" "enabled-apis" {
  project = "${google_project.project.project_id}"
  services = ["compute.googleapis.com"]
}

data "google_compute_default_service_account" "default" {
  project = "${google_project.project.project_id}"
}

output "default_account" {
  value = "${data.google_compute_default_service_account.default.email}"
}
```

However it results in this during the plan:

```
* module.project-infrastructure-sandbox.output.default_account: Resource 'data.google_compute_default_service_account.default' does not have attribute 'email' for variable 'data.google_compute_default_service_account.default.email'
```

So it looks like the output variable is trying to read `.email` from the data resource during the plan.

Any ideas?

**Comment by emilymye** [comment]
Have you tried adding a "depends_on" block to the data source? I'll have to try it out but I'm hoping that's the easy fix.  

i.e.

```
data "google_compute_default_service_account" "default" {
    depends_on = ["google_project.project"] 
    project = "${google_project.project.project_id}"
}
```

**Comment by Stono** [comment]
Yeah tried that, didn't work :(

**Comment by shraddhajain01** [comment]
I am facing the same issue. @Stono Did you find any workaround for it?

**Comment by shraddhajain01** [comment]
Anyone knows any workaround for this? I really want to create the above resources in the sequence mentioned by @Stono. Thanks.

**Comment by Stono** [comment]
Yes the latest version of the provider fixes it #1690
